lxd/resource_lxd_network_zone.go

```diff
@@ -0,0 +1,241 @@
+package lxd
+
+import (
+	"log"
+
+	"github.com/canonical/lxd/shared/api"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceLxdNetworkZone() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLxdNetworkZoneCreate,
+		Update: resourceLxdNetworkZoneUpdate,
+		Delete: resourceLxdNetworkZoneDelete,
+		Exists: resourceLxdNetworkZoneExists,
+		Read:   resourceLxdNetworkZoneRead,
+		Importer: &schema.ResourceImporter{
+			State: resourceLxdNetworkZoneImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"remote": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "",
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"config": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceLxdNetworkZoneCreate(d *schema.ResourceData, meta interface{}) error {
+	p := meta.(*lxdProvider)
+	server, err := p.GetInstanceServer(p.selectRemote(d))
+	if err != nil {
+		return err
+	}
+
+	if v, ok := d.GetOk("project"); ok && v != "" {
+		project := v.(string)
+		server = server.UseProject(project)
+	}
+
+	name := d.Get("name").(string)
+	desc := d.Get("description").(string)
+	config := resourceLxdConfigMap(d.Get("config"))
+
+	log.Printf("[DEBUG] Creating network zone %s with config: %#v", name, config)
+	req := api.NetworkZonesPost{Name: name}
+	req.Config = config
+	req.Description = desc
+
+	mutex.Lock()
+	err = server.CreateNetworkZone(req)
+	mutex.Unlock()
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(name)
+
+	return resourceLxdNetworkZoneRead(d, meta)
+}
+
+func resourceLxdNetworkZoneRead(d *schema.ResourceData, meta interface{}) error {
+	p := meta.(*lxdProvider)
+	server, err := p.GetInstanceServer(p.selectRemote(d))
+	if err != nil {
+		return err
+	}
+
```

Comment:
Same comment as above. I think adding extension check to Read and Create should be sufficient.

Reply:
Done.